ui/app/models/cluster.js

```diff
@@ -49,47 +49,17 @@ export default DS.Model.extend({
 
   anyReplicationEnabled: or('{dr,performance}.replicationEnabled'),
 
-  stateDisplay(state) {
-    if (!state) {
-      return null;
+  modeState: computed(
+    'dr.mode',
+    'performance.mode',
+    'replicationMode',
+    'dr.state',
+    'performance.state',
+    function() {
+      const mode = this.replicationMode;
+      return this.get(`${mode}.state`);
     }
-    const defaultDisp = 'Synced';
-    const displays = {
-      'stream-wals': 'Streaming',
-      'merkle-diff': 'Determining sync status',
-      'merkle-sync': 'Syncing',
-    };
-
-    return displays[state] || defaultDisp;
-  },
-
-  drStateDisplay: computed('dr.state', function() {
-    return this.stateDisplay(this.get('dr.state'));
-  }),
-
-  performanceStateDisplay: computed('performance.state', function() {
-    return this.stateDisplay(this.get('performance.state'));
-  }),
-
-  stateGlyph(state) {
-    const glyph = 'check-circle-outline';
```

Comment:
this is the default glyph to display, including when state is unknown. i let this be, but maybe we shouldn't show an icon at all if state can't be found?